### PR TITLE
Add component for search

### PR DESCRIPTION
**Pull request by SebiLimbu**

### CODE REVIEW

File: src/components.js [hunk 1]
```diff
@@ -2,10 +2,11 @@ import Footer from './components/Footer';
 import CategoryListItem from './components/CategoryListItem';
 import NavBar from './components/NavBar';
 import ServicePane from './components/ServicePane';
-
+import Search from './components/Search';
 export default {
```
Comment (danni) [inline src/components.js:11]: Keep a space here.

**Comment by danni** [comment]
Make sure you are passing the style tests. You can symlink `.githooks` to `.git/hooks` to have them run when you commit.

Write a BDD test for the behaviour you wish to validate. It's best to write the behavioural test first so you know what you're writing. Something like:

```
When I visit the Ask Izzy app
And I do a search for "dog food"
Then I am taken to the page "Search Results: dog food"
And I see 3 results
```

You will need to define most of these steps.

File: src/components.js (hunk 1 above)
Comment (SebiLimbu) [inline src/components.js:11]: Style test passed.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@
 /public/static/*
 npm-debug.log
 .env
+dev-server
```
Comment (DanielHeath) [inline .gitignore]: I don't think this is right

File: script/check-shrinkwrap [hunk 3]
```diff
@@ -15,7 +15,9 @@ var shrinkwrapDeps = Object.keys(shrinkwrapJSON.dependencies).sort();
 
 // All deps in node_modules
 var node_modules = fs.readdirSync("node_modules").sort();
-node_modules.splice(0, 1); // Remove '.bin' - everything else in node_modules is a module.
+while (node_modules[0][0] === ".") {
```
Comment (DanielHeath) [inline script/check-shrinkwrap]: I'll pull this out into master

File: src/components/Search.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/* @flow */
+import React from "react";
+
+class Search extends React.Component {
+
+    render(): React.Element {
+        return (
+                <div className="search-container">
```
Comment (DanielHeath) [inline src/components/Search.js]: This looks like it's been double indented

File: src/components/ServicePane.js [hunk 5]
```diff
@@ -63,6 +63,7 @@ export default class ServicePane extends React.Component {
                 zDepth={1}
             >
                 <div style={{padding:10}}>
+
```
Comment (DanielHeath) [inline src/components/ServicePane.js]: I'm not sure why this is added

File: src/pages/HomePage.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ import Router, {RouteHandler} from 'react-router';
 import NavBar from "../components/NavBar";
 import Footer from "../components/Footer";
 import mui from "material-ui";
-
+import Search from "../components/Search";
```
Comment (DanielHeath) [inline src/pages/HomePage.js:5]: Keep a space here

File: src/styles/normalize.css [hunk 7]
```diff
@@ -422,3 +422,26 @@ td,
 th {
   padding: 0;
 }
+
```
Comment (DanielHeath) [inline src/styles/normalize.css]: I don't think search-related code belongs in the normalize css

File: src/styles/normalize.css [hunk 8]
```diff
@@ -422,3 +422,26 @@ td,
...
+.search-container{
+  /*margin-left: 5%;
```
Comment (DanielHeath) [inline src/styles/normalize.css]: Do we need to keep commented-out code?

**Comment by DanielHeath** [comment]
Not sure why  `when_i_search_for_shelter.png` is added

File: src/components/Search.scss [hunk 9]
```diff
@@ -0,0 +1,23 @@
+@import '../styles/colors.scss';
+
+/**
+*Sets the div container for the input (search) element.
+**/
+.search-container{
+  width: 80%;
+  margin: 0 auto;
+}
+
```
Comment (DanielHeath) [inline src/components/Search.scss:13]: We probably dont need that much whitespace between rules - one line should be fine

File: src/components/Search.scss [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+
+/**
+*Setting the layout of the search textbox with search icon
+**/
+#search-box{
+  width: 100%; /*padding: .5em 1em;*/
```
Comment (DanielHeath) [inline src/components/Search.scss]: Avoid checking in commented out code unless there's a good reason to do so

**Comment by DanielHeath** [comment]
I've rebased this PR against master. @SebiLimbu - you'll need to reset your local copy.

**Pull request closed without merging**